src/main/java/uk/gov/hmcts/juror/api/moj/service/PoolCreateServiceImpl.java

```diff
@@ -420,24 +421,28 @@ private List<JurorPool> getJurorPools(String login, String owner, PoolCreateRequ
                 // Increment the previous sequence number by one to get the new sequence number
                 sequenceNumber++;
 
-                if (!Objects.equals(jurorPool.getStatus().getStatus(), IJurorStatus.DISQUALIFIED)) {
-                    // create a summons letter for juror
-                    printDataService.printSummonsLetter(jurorPool);
-                }
-
                 if (jurorsFound == poolCreateRequestDto.getCitizensToSummon()) {
                     break;  // we've found the number of jurors required, no need to process any further.
                 }
 
             }
 
+            // Saving records (bulk)
+            jurorRepository.saveAll(jurorPools.stream().map(JurorPool::getJuror).toList());
+            jurorPoolRepository.saveAll(jurorPools);
+
+            // create a summons letter for juror
+            printDataService.bulkPrintSummonsLetter(jurorPools.stream()
+                .filter(jurorPool -> !Objects.equals(jurorPool.getStatus().getStatus(), IJurorStatus.DISQUALIFIED))
+                .toList());
```

Comment:
it maybe worth creating an integration test here where we try and save an empty list... I have a feeling this may fail (But I could also be vastly incorrect)

Same for bulk print summons letter

Reply:
You were correct, but further testing the code it can never reach that point in it's current state. I've added a test to the create pool controller to verify that, also a bulk print summons letter tests for null list already exists.